apis/python/src/tiledb/vector_search/index.py

```diff
@@ -37,15 +39,32 @@ def __init__(
         self.update_arrays_uri = None
         self.index_version = self.group.meta.get("index_version", "")
 
-
     def query(self, queries: np.ndarray, k, **kwargs):
-        updated_ids = set(self.read_updated_ids())
-        retrieval_k = k
-        if len(updated_ids) > 0:
-            retrieval_k = 2*k
-        internal_results_d, internal_results_i = self.query_internal(queries, retrieval_k, **kwargs)
         if self.update_arrays_uri is None:
-            return internal_results_d[:, 0:k], internal_results_i[:, 0:k]
+            return self.query_internal(queries, k, **kwargs)
+
+        # Query with updates
+        # Perform the queries in parallel
+        retrieval_k = 2 * k
+        kwargs["nthreads"] = int(mp.cpu_count() / 2)
+        parent_conn, child_conn = mp.Pipe()
+        p = mp.Process(
```

Comment:
Is there a reason not to use threading here? That will be a bit safer -- with with this, we'll need to call `mp.set_start_method('spawn')` to avoid issues with the default (`fork`) on linux.